test/core/filters/client_auth_filter_test.cc

```diff
@@ -140,72 +115,36 @@ TEST_F(ClientAuthFilterTest, CreateFailsWithoutRequiredChannelArgs) {
 }
 
 TEST_F(ClientAuthFilterTest, CreateSucceeds) {
-  auto filter = ClientAuthFilter::Create(MakeChannelArgs(absl::OkStatus()),
-                                         ChannelFilter::Args());
+  auto filter = MakeChannel(MakeChannelArgs(absl::OkStatus()));
   EXPECT_TRUE(filter.ok()) << filter.status();
 }
 
 TEST_F(ClientAuthFilterTest, CallCredsFails) {
-  auto filter = ClientAuthFilter::Create(
-      MakeChannelArgs(absl::UnauthenticatedError("access denied")),
-      ChannelFilter::Args());
-  // TODO(ctiller): use Activity here, once it's ready.
-  TestContext<Arena> context(arena_.get());
-  TestContext<grpc_call_context_element> promise_call_context(call_context_);
-  auto promise = filter->MakeCallPromise(
-      CallArgs{ClientMetadataHandle(&initial_metadata_batch_,
-                                    Arena::PooledDeleter(nullptr)),
-               nullptr, nullptr, nullptr},
-      [&](CallArgs /*call_args*/) {
-        return ArenaPromise<ServerMetadataHandle>(
-            [&]() -> Poll<ServerMetadataHandle> {
-              return ServerMetadataHandle(&trailing_metadata_batch_,
-                                          Arena::PooledDeleter(nullptr));
-            });
-      });
-  auto result = promise();
-  ServerMetadataHandle* server_metadata =
-      absl::get_if<ServerMetadataHandle>(&result);
-  ASSERT_TRUE(server_metadata != nullptr);
-  auto status_md = (*server_metadata)->get(GrpcStatusMetadata());
-  ASSERT_TRUE(status_md.has_value());
-  EXPECT_EQ(*status_md, GRPC_STATUS_UNAUTHENTICATED);
-  const Slice* message_md =
-      (*server_metadata)->get_pointer(GrpcMessageMetadata());
-  ASSERT_TRUE(message_md != nullptr);
-  EXPECT_EQ(message_md->as_string_view(), "access denied");
+  //  initial_metadata_batch_.Set(HttpAuthorityMetadata(), target_.Ref());
+  StrictMock<Call> call(MakeChannelWithCallCredsResult(
+      absl::UnauthenticatedError("access denied")));
+  call.Start(call.NewClientMetadata({{":authority", target()}}));
+  EXPECT_CALL(
+      call, Finished(AllOf(
+                HasMetadataKeyValue(
```

Comment:
Failing a call with a non-OK status is likely to be a sufficiently common case that I would suggest adding a matcher just for that case.  It would be nice to say something like `CallStatusIs(absl::UnauthenticatedError("access denied"))`.